src/main/java/org/opensearch/knn/jni/JNIService.java

```diff
@@ -211,6 +212,32 @@ public static long loadIndex(String indexPath, Map<String, Object> parameters, K
         );
     }
 
+    /**
+     * Load an index via Lucene's IndexInput.
+     *
+     * @param readStream A wrapper having Lucene's IndexInput to load bytes from a file.
+     * @param parameters Parameters to be used when loading index
+     * @param knnEngine  Engine to load index
+     * @return Pointer to location in memory the index resides in
+     */
+    public static long loadIndex(IndexInputWithBuffer readStream, Map<String, Object> parameters, KNNEngine knnEngine) {
+        if (KNNEngine.NMSLIB == knnEngine) {
+            throw new UnsupportedOperationException("Loading NMSLIB with a read stream is not supported at the moment");
+        }
```

Comment:
Had it here for being conservative. 
Also I thought it would helpful for the reader to have a easier understanding why there's only FAISS in this method.
But, yeah! there won't be NMSLIB to be passed down to here.

Reply:
The thing is the last throw is just dead code as it stands now. Would recommend removing if its handled correctly in favor of reducing code since there is no ambiguity here

Reply:
Not sure I'm understanding `The thing is the last throw is just dead code as it stands now. `.
When we throw, the thrown exception will be eaten silently?
Hmm... I think KNNWeight would trigger this, no?

Reply:
I was talking about this, I will leave it up to you on this. not a blocker

```
        throw new IllegalArgumentException(
            String.format(Locale.ROOT, "LoadIndex not supported for provided engine : %s", knnEngine.getName())
        );
```